libs/execution_context/ebpf_program.c

```diff
@@ -179,12 +179,14 @@ _ebpf_program_program_info_provider_changed(
             if (return_value != EBPF_SUCCESS)
                 goto Exit;
 
+#if !defined(EBPF_JIT_ALWAYS_ON)
```

Comment:
I think we also need to change `ebpf_verify_and_load_program()` function so that it returns error when `EBPF_JIT_ALWAYS_ON` is defined and client passes `EBPF_EXECUTION_INTERPRET` execution type.

Reply:
Would it be ok if the EC just returned EBPF_BLOCKED_BY_POLICY if EBPF_JIT_ALWAYS_ON is enabled but interpret is requested?

Reply:
I would say yes.

Reply:
I think yes (as long as we gracefully fail the program load request).

Another thing is, do we need to add any tests (or modify existing tests -- if we are enforcing JIT for release build, all the test cases for INTERPRET mode will start failing)?

Reply:
Agree I expect some tests to change.